Sources/Extensions/NSDictionary+Toggles.swift

```diff
@@ -0,0 +1,39 @@
+//  NSDictionary+Toggles.swift
+
+import Foundation
+
+extension NSDictionary {
+    public enum ToggleConversionError: Error {
+        case nsDictionaryInvalidFormat
+        case nsDictionaryEmpty
+    }
+    
+    public func convertToToggleDataSource() throws -> [Variable: Toggles.Value] {
```

Comment:
Would `convertToTogglesDataSource` (plural) sound better as the library is named Toggles?